### PR TITLE
Netcode Fix: Boss Summon Item issue fixes

### PR DESCRIPTION
**Pull request by Flowaria**
Resolved Issue:
- Every boss summon items:
  - Removed `Non-Owner-Client` -> `Server` Netpath to reduce network usage
  - Added Util method for spawning boss: `CalamityUtils.SpawnBossUsingItem`, `CalamityUtils.SpawnBossOnPosUsingItem`
- `DeathWhistle`, `ProfandedCore`, `DecapoditaSprout`, `AstralChunk`:
  - now always spawn boss in intended position while in MP
- `OldPowerCell`:
  - now always can spawn Golem
  - Lihzahrd Temple check condition has replaced to `player.ZoneLihzhardTemple` instead of checking surrounding blocks
- `EidolonTablet`:
  - now properly spawn Cultist on MP
  - Lunar Event condition check now works in MP

### CODE REVIEW

File: Items/SummonItems/EidolonTablet.cs [hunk 1]
```diff
@@ -34,21 +33,48 @@ public override void ModifyResearchSorting(ref ContentSamples.CreativeHelper.Ite
...
+                    case NPCID.CultistTablet:
+                    case NPCID.CultistBoss:
+                    case NPCID.LunarTowerSolar:
+                    case NPCID.LunarTowerVortex:
+                    case NPCID.LunarTowerNebula:
+                    case NPCID.LunarTowerStardust:
+                    case NPCID.MoonLordCore:
+                        return false;
+                }
+            }
```
Comment (Ozzatron) [inline Items/SummonItems/EidolonTablet.cs]: This loop should be looping over ActiveNPCs instead of using a hardcoded 200.

**Review by Ozzatron — CHANGES_REQUESTED** [review]
Almost perfect, except for the dirty Cultist loop. Please see that comment.

**Review by Ozzatron — APPROVED** [review]
Thanks for fixing the loop. That was fast.

**Review by mountaindrew10 — APPROVED** [review]
Everything looks good, thank you.

**Pull request merged**